zebra/zebra_rib.c

```diff
@@ -2693,6 +2693,7 @@ static void early_route_memory_free(struct zebra_early_route *ere)
 	if (ere->re_nhe)
 		zebra_nhg_free(ere->re_nhe);
 
+	zapi_re_opaque_free(ere->re->opaque);
```

Comment:
maybe it'd be good to take several of these common "re" memory free-ers and put them in one common function?

Reply:
I spent a few minutes trying to figure out how to make this better and frankly the whole thing needs to be reworked, which is far more than I want to do at this point in time.  This function early_route_memory_free  is already abstracted to handle all the freeing points in early_route_add/delete.  Let's try to get the address sanitizer problem fixed and I will come back around in the future after I have some time to rework the whole thing